### PR TITLE
don't load files until sources() or dependencies() is called

### PR DESCRIPTION
**Pull request by FredKSchott**
Brought up in: https://github.com/Polymer/polymer-build/issues/97#issuecomment-268619992

Currently, files begin loading from disk the second the PolymerProject instance is created. This is problematic for a build library because the files you need to load may not exist at the start of the build. They could be moved or even generated as a part of the build process, before polymer-build is started. 

It's the setup of a listener on the vinyl source stream (`_sourcesStream`) that starts the actual loading. So this PR moves all stream setup from the constructor (at `PolymerProject` creation-time)  to the time of their first use. 

Fixing this was an original goal for https://github.com/Polymer/polymer-build/pull/77, but it was dropped to keep that PR scope small. 

/cc @justinfagnani @rictic @usergenic

### CODE REVIEW

File: src/test/analyzer_test.ts [hunk 1]
```diff
@@ -232,16 +233,63 @@ suite('Analyzer', () => {
     });
 
     const analyzer = new BuildAnalyzer(config);
+    assert.isFalse(analyzer.started);
+    analyzer.sources().pipe(new NoopStream());
+    assert.isTrue(analyzer.started);
+  });
+
+
+  test('calling dependencies() starts analysis', () => {
```
Comment (FredKSchott) [inline src/test/analyzer_test.ts]: These three tests (two new and 1 updated) are the only testing changes made by me. The rest of this `test/` diff is due to clang-formatting after the `.sources -> .sources()` & `dependencies -> dependencies()` change.

**Comment by justinfagnani** [comment]
Isn't there a way for a ReadableStream implementor to be notified when it has 'data' listeners? We don't want to start producing data when someone _requests_ one of the file streams, but when they've hooked it up to a destination.

File: src/analyzer.ts [hunk 2]
```diff
@@ -198,15 +214,17 @@ export class BuildAnalyzer {
    * Return _dependenciesOutputStream, which will contain fully loaded file
    * objects for each dependency after analysis.
    */
-  get dependencies(): NodeJS.ReadableStream {
+  dependencies(): NodeJS.ReadableStream {
```
Comment (FredKSchott) [inline src/analyzer.ts]: startAnalysis() begins the loading of files from disk. This was a significant enough side effect to warrant the change. And as a little bonus, it also now matches the `polymerProject.dependencies()` method that most users interact with.

**Comment by FredKSchott** [comment]
@justinfagnani not that I've been able to find. The closest is the `pipe` event, but that is emitted on the destination stream and not the source stream. While I would prefer that behavior as well, the only way that I can see is by extending the Stream class to add our own `pipe()` logic, or by making the user call `startAnalysis()`/`start()`  explicitly to start the stream. 

But the only way to run into the issue you're describing is to call `polymerProject.sources()`/`polymerProject.dependencies()` and then not `pipe()` or connect the stream on that same tick. I've never seen any of our users do this before, and I can't think of a scenario where you'd ever need to. Given that, I don't think this is worth prioritizing right now (or at the very least, I don't think it is worth blocking this improvement over the current logic).

**Comment by FredKSchott** [comment]
@justinfagnani @rictic PTAL

**Comment by justinfagnani** [comment]
I'm not suggesting blocking this, just wondering if it's possible given that that's the behavior we want.

File: src/analyzer.ts (hunk 2 above)
Comment (justinfagnani) [inline src/analyzer.ts]: I agree with that reasoning there, even though the breaking change and churn is unfortunate.

Comment (justinfagnani) [inline src/analyzer.ts]: Oh, I see this is on analyzer anyway, which isn't often the public interface. I take back the churn comment.

**Pull request merged**